### PR TITLE
Updated specification for JSON requests to daemon

### PR DESCRIPTION
**Pull request by Nathan-Hausman**
Merge new documentation for JSON requests.

### CODE REVIEW

**Comment by Nathan-Hausman** [comment]
New specifications as follows:

```jsonc
{
  "repo_url": "https://the.repo",
  "repo_branch": null || "thebranch",

  // ONE of the following
  "deploy_config": null,   // use "deploy_config_path" (must not be null) 
  "deploy_config": { /* JSON of deployment config */ },
  "deploy_config": "{ JSON string of deployment config }",
  
  // ONE of the following
  "service_config": null,   // use "service_config_path" (must not be null) 
  "service_config": { /* JSON of service config */ },
  "service_config": "{ JSON string of service config }",

  // ONE of the following
  "deploy_config_path": null,   // use "deploy_config" (must not be null) 
  "deploy_config_path": "path/to/config",

  // ONE of the following
  "service_config_path": null,   // use "service_config" (must not be null) 
  "service_config_path": "path/to/config",
}
```

File: documentation/build-service.md [hunk 1]
```diff
@@ -40,9 +40,22 @@ Make a `POST` request to `<host>:8800/build` with the following JSON content:
...
+  "service_config": { /* JSON of service config */ },
+  "service_config": "{ JSON string of service config }",
+
+  // ONE of the following
+  "deploy_config_path": null,   // use "deploy_config" (must not be null) 
+  "deploy_config_path": "path/to/config",
+
+  // ONE of the following
+  "service_config_path": null,   // use "service_config" (must not be null) 
+  "service_config_path": "path/to/config",
```
Comment (detjensrobert) [inline documentation/build-service.md]: We can make this simpler and combine the `_config` and `_config_path` options:

```jsonc
  // ONE of the following
  "deploy_config": null,  // use default path
  "deploy_config": "path/to/deploy.yml",
  "deploy_config": { /* custom JSON config object */ },
  
  // ONE of the following
  "service_config": null,  // use default path
  "service_config": "path/to/deploy.yml",
  "service_config": { /* custom JSON config object */ },
```

Thoughts? 

What should the default path be? `/kaas.{deploy,config}.yml`?

Comment (Nathan-Hausman) [reply documentation/build-service.md]: That seems perfect. And sure lets go with that as the default for now.

**Comment by Nathan-Hausman** [comment]
Changed documentation to what Robert recommended.

**Comment by detjensrobert** [comment]
cc @Nathan-Hausman for review

**Review by Nathan-Hausman — COMMENTED** [review]
Looks good to me.

**Pull request merged**